RoaringBitmap/src/main/java/org/roaringbitmap/longlong/Roaring64Bitmap.java

```diff
@@ -863,10 +864,12 @@ public void advanceIfNeeded(long minval) {
         }
       }
 
-      // advance inner
-      char low = LongUtils.lowPart(minval);
-      charIterator.advanceIfNeeded(low);
-      hasNextCalled = false;
+      if (Arrays.equals(this.high, minHigh)) {
```

Comment:
seems no need to add this additional conditional judgement

Reply:
I agree that commenting out the IF check, the tests still pass. And ponder if the check was to be left, if the `Arrays.equals` should be replace with the `compareHigh` to be consistent.

Reply:
scratch my high comment here, I just observed that the compareHigh are doing >= or <= in the forward/reverse iterators, where this was just looking for equality.

Reply:
also as I rewrite the code to understand what it's doing, the `IF` here (line 867) is needed as the compareHigh on line 842 will branch into this line when we step over the block we are looking for at which point we want the first value, not to step forward for the lower parts on the wrong upper block.

Reply:
Yes, we definitely need this. This is what caused the first bug @simeonpilgrim originally reported.